src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,166 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> head;
+    private Node<T> tail;
+    private int size;
+
     @Override
     public void add(T value) {
+        linkLast(value);
     }
 
     @Override
     public void add(T value, int index) {
+        checkIndexInBounds(index);
+
+        if (size == index) {
+            linkLast(value);
+        } else {
+            Node<T> nodeAtIndex = getNodeIndex(index);
+            Node<T> prevNode = nodeAtIndex.prev;
+            Node<T> newNode = new Node<>(prevNode, value, nodeAtIndex);
+            nodeAtIndex.prev = newNode;
+
+            if (prevNode == null) {
+                head = newNode;
+            } else {
+                prevNode.next = newNode;
+            }
+            size++;
+        }
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (int i = 0; i < list.size(); i++) {
+            add(list.get(i));
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        checkIndexInBounds(index);
+        checkIndexAtSizeRange(index);
+        return getNodeIndex(index).value;
```

Comment:
![image](https://user-images.githubusercontent.com/20326858/224541558-326ec017-8940-4a77-aa0b-b7077a97ee93.png)

**This is code repeating**

![image](https://user-images.githubusercontent.com/20326858/224541580-6cb78f49-32b1-4c11-b71a-501565b59e39.png)